File: caslogin/encrypt.js
const window={}
require('./jsencrypt').myfunc(window)
const encrypt = new window.JSEncrypt()
encrypt.setPublicKey('MFwwDQYJKoZIhvcNAQEBBQADSwAwSAJBAJ7aKWURpJx4m8i4pB9P2pzNriT3UyhK6H20meKv8gFJmERA482JSi/DvZ4SkxP9INL5h8lGGvu5W3eTrpJaN3MCAwEAAQ==')

function strEnc(text)
{
    return encrypt.encrypt(text)
}

File: caslogin/encrypt.py
import execjs
import re
import os

def encrypt(username, password, public_key):
    root_dir = os.path.dirname(__file__)
    with open(os.path.join(root_dir, "encrypt.js"), "r") as js_file:
        jscode = js_file.read()
    jscode = re.sub(r"setPublicKey\('(.*?)'\)", f"setPublicKey('{public_key}')", jscode)
    encrypt_js = execjs.compile(jscode)
    ul = encrypt_js.call("strEnc", username)
    pl = encrypt_js.call("strEnc", password)
    return ul, pl
